src/masternodes/rpc_vault.cpp

```diff
@@ -0,0 +1,609 @@
+            }
+            collValue = ValueFromAmount(totalCollateral);
+            loanValue = ValueFromAmount(totalLoan);
+
+            UniValue collateralBalances{UniValue::VARR};
+            UniValue loanBalances{UniValue::VARR};
+
+            if (collaterals)
+                collateralBalances = AmountsToJSON(collaterals->balances);
+
+            if (auto loanTokens = pcustomcsview->GetLoanTokens(vaultId)){
+                TAmounts balancesInterest{};
+                for (const auto& loan : loanTokens->balances) {
+                    auto rate = pcustomcsview->GetInterestRate(vault.schemeId, loan.first);
+                    CAmount value = loan.second + MultiplyAmounts(loan.second, TotalInterest(*rate, ::ChainActive().Height()));
+                    balancesInterest.insert({loan.first, value});
+                }
+                loanBalances = AmountsToJSON(balancesInterest);
+            }
+
+            result.pushKV("collateralAmounts", collateralBalances);
+            result.pushKV("loanAmount", loanBalances);
+            result.pushKV("collateralValue", collValue);
+            result.pushKV("loanValue", loanValue);
+            result.pushKV("currentRatio", (int)ratio);
+        }
+        return result;
+    }
+}
+
+UniValue createvault(const JSONRPCRequest& request) {
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    RPCHelpMan{"createvault",
+                "Creates a vault transaction.\n" +
+                HelpRequiringPassphrase(pwallet) + "\n",
+                {
+                    {"ownerAddress", RPCArg::Type::STR, RPCArg::Optional::NO, "Any valid address or \"\" to generate a new address"},
+                    {"loanSchemeId", RPCArg::Type::STR, RPCArg::Optional::OMITTED,
+                        "Unique identifier of the loan scheme (8 chars max). If empty, the default loan scheme will be selected (Optional)"
+                    },
+                    {"inputs", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG,
+                        "A json array of json objects",
+                            {
+                                {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                     {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                     {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                   "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                   HelpExampleCli("createvault", "") +
+                   HelpExampleCli("createvault", "" "LOAN0001") +
```

Comment:
```suggestion
                   HelpExampleCli("createvault", "\"\" LOAN0001") +
```

Reply:
Is there others like that?

Reply:
I've just seen this one so far but will let you know if I see some others.